Frame.cc

```diff
@@ -961,16 +961,20 @@ bool Frame::FillSpectralProfileData(
                         profile_ok = true;
                         return profile_ok;
                     }
+                    bool region_mask_ok(false);
                     bool use_swizzled_data(false);
+                    std::unique_lock<std::mutex> guard(_image_mutex);
                     try {
                         // check is the region mask valid (outside the lattice or not)
                         region->XyMask();
```

Comment:
why are we calling `XyMask()` twice? @confluence, did you put this in? Getting the mask might be expensive, since it involves per-pixel testing of polygons. Its performance is likely to be _O(A)_, where _A_ is the area of the bounding box of the region

Reply:
Actually is me. Since `XyMask()` may throw an exception, then it will cause the hanging issue if a region is outside the lattice.

Reply:
Actually, given that it seems to just return a reference to the mask lattice, I don't think it's a problem.

Reply:
might make more sense to reuse the first result of `region->XyMask()`, otherwise it's quite hard to understand what's going on.

Reply:
Another option would be to check the region bounding box first, perhaps?

Reply:
I think we can call `region->XyMask()` only once. Have submitted a new commit.